server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -84,13 +88,34 @@ public void beforeIndexShardClosed(ShardId shardId, @Nullable IndexShard indexSh
         }
     }
 
+    /**
+     * Returns the Latest checkpoint received by replica shard based on shard ID.
+     * Returns null if replica shard has not received a checkpoint before
+     * @param shardId      Shard Id of replica shard
+     */
+    private ReplicationCheckpoint getLatestReceivedCheckpoint(ShardId shardId) {
```

Comment:
Nitpick - this method seems unnecessary since `Map.get` will return `null` if no value is found for the specified key. You can just use `latestReceivedCheckpoint.get(shardId)` inline.

Reply:
+1